### PR TITLE
Feature/configurable state pivot distance

### PR DESCRIPTION
**Pull request by asdacap**
- Was originally want to add config flag to configure the max and min distance of state pivot from head. 
- Then changed a lot of unit test.
- Moved `SyncPeerPool` to dependency injection.
- StopAsync of ISyncPeerPool and ISynchronizer move to use `IAsyncDisposable`. 
- Ensure all container and therefore peerpool/synchronizer is disposed in unit tests.
- Seems to redure the hanging synchronization package a bit.

#### What types of changes does your code introduce?

- [X] New feature (a non-breaking change that adds functionality)

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [X] Yes
- [ ] No

#### Notes on testing

- Testing...

### CODE REVIEW

**Comment by LukaszRozmej** [comment]
LOL just had a meeting with @damian-orzechowski when we discussed it can help with optimism sync not go to to latest head for pivot for state

File: src/Nethermind/Nethermind.Synchronization.Test/FastSync/StateSyncFeedHealingTests.cs
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Synchronization.Test/FastSync/StateSyncFeedHealingTests.cs:141]: `await using`?

**Review by LukaszRozmej — APPROVED** [review]
Looks great for me, wait for @damian-orzechowski review before merging.

File: src/Nethermind/Nethermind.Synchronization.Test/FastSync/StateSyncFeedHealingTests.cs
Comment (asdacap) [reply src/Nethermind/Nethermind.Synchronization.Test/FastSync/StateSyncFeedHealingTests.cs]: Fixed

File: src/Nethermind/Nethermind.Synchronization/SnapSync/Pivot.cs
Comment (damian-orzechowski) [inline src/Nethermind/Nethermind.Synchronization/SnapSync/Pivot.cs:72]: Add `if (_logger.IsInfo)` ?

**Review by damian-orzechowski — APPROVED** [review]

**Pull request merged**
